cybersyn/scripts/central-planning.lua

```diff
@@ -9,6 +9,36 @@ local band = bit32.band
 local table_remove = table.remove
 local random = math.random
 
+local HASH_STRING = "CS_QUALITY_SEP"
+
+---@param name string The name of the item
+---@param quality string The name of the quality of the item or nil if it is common
+---@return string
+function hash_item(name, quality)
+	if quality == nil then
+		return name
+	else
+		return name .. HASH_STRING .. quality
+	end
+end
+
+---@param sig SignalId
+---@return string
+function hash_signal(sig)
+	return hash_item(sig.name, sig.quality)
+end
+
+---@param hash string
+---@return SignalId
+function unhash_signal(hash)
+	local index = string.find(hash, HASH_STRING)
+
+	if index == nil then
+		return { name = hash }
```

Comment:
I think it would be more useful to return two values, i.e. `return name, quality` (where quality can be nil)

Reply:
Adressed in 1e2819aed231d1a5a998be2c05ae00c5ada00e9d